qiita_pet/handlers/software.py

```diff
@@ -144,6 +149,37 @@ def _default_parameters_parsing(node):
 
         wparams = w.parameters
 
+        # adding nodes without edges
+        # as a first step if not_used_nodes is not empty we'll confirm that
+        # nodes/edges are empty; in theory we should never hit this
+        if not_used_nodes and (nodes or edges):
+            raise ValueError(
+                'Error, please check your workflow configuration')
+
+        # note that this block is similar but not identical to adding connected
+        # nodes
+        for i, (_, x) in enumerate(not_used_nodes.items()):
+            vals_x, input_x, output_x = _default_parameters_parsing(x)
+            if at in input_x[0][1]:
+                input_x[0][1] = at
+            else:
+                input_x[0][1] = '** WARNING, NOT DEFINED **'
```

Comment:
Are we sure that it wouldn't be better to raise an Error if something was undefined instead of encoding a warning message?

Reply:
I'm 100% sure about it.

Reply:
ok.